ArduCopter/mode_auto.cpp

```diff
@@ -992,7 +992,10 @@ void ModeAuto::takeoff_run()
     // if the user doesn't want to raise the throttle we can set it automatically
     // note that this can defeat the disarm check on takeoff
     if ((copter.g2.auto_options & (int32_t)Options::AllowTakeOffWithoutRaisingThrottle) != 0) {
-        copter.set_auto_armed(true);
+        if (motors->armed() && copter.ap.auto_armed == false) {
```

Comment:
nitpick: could we make this !copter.ap.auto_armed?

Reply:
sure